src/sql/workbench/contrib/notebook/browser/notebookViews/insertCellsModal.ts

```diff
@@ -0,0 +1,326 @@
+	}
+
+	public onOptionChecked(optionName: string) {
+		this.viewModel.setOptionValue(optionName, (<Checkbox>this._optionsMap[optionName]).checked);
+		this.validate();
+	}
+
+	private getOptions(): ServiceOption[] {
+		const activeView = this._context.getActiveView();
+		const cellsAvailableToInsert = activeView.hiddenCells;
+		return cellsAvailableToInsert.map((cell) => ({
+			name: cell.cellGuid,
+			displayName: cell.renderedOutputTextContent[0]?.substr(0, 20) ?? localize("insertCellsModal.untitled", "Untitled Cell : {0}", cell.cellGuid),
+			description: '',
+			groupName: undefined,
+			valueType: ServiceOptionType.boolean,
+			defaultValue: '',
+			objectType: undefined,
+			categoryValues: [],
+			isRequired: false,
+			isArray: undefined,
+		}));
+	}
+
+	public override render() {
+		super.render();
+
+		this._submitButton = this.addFooterButton(localize('insertCellsModal.Insert', "Insert"), () => this.onSubmitHandler());
+		this._cancelButton = this.addFooterButton(localize('insertCellsModal.Cancel', "Cancel"), () => this.onCancelHandler(), 'right', true);
+
+		this._register(attachButtonStyler(this._submitButton, this._themeService));
+		this._register(attachButtonStyler(this._cancelButton, this._themeService));
+
+		attachModalDialogStyler(this, this._themeService);
+		this.validate();
+	}
+
+	private validate() {
+		if (Object.keys(this._optionsMap).length) {
+			this._submitButton.enabled = true;
+		} else {
+			this._submitButton.enabled = false;
+		}
+	}
+
+	private onSubmitHandler() {
+		this.viewModel.restoreCells();
+		this.close();
+	}
+
+	private onCancelHandler() {
+		this.close();
+	}
+
+	public close(): void {
+		return this.hide();
+	}
+
+	public async open(): Promise<void> {
+		this.show();
+	}
+
+	public override dispose(): void {
+		super.dispose();
+		for (let key in this._optionsMap) {
+			let widget = this._optionsMap[key];
+			widget.dispose();
+			delete this._optionsMap[key];
+		}
+	}
+
+	public async generateScreenshot(cell: ICellModel, screenshotWidth: number = 300, screenshowHeight: number = 300, backgroundColor: string = '#ffffff'): Promise<any> {
+		let componentFactory = this._componentFactoryResolver.resolveComponentFactory(TextCellComponent);
+		let component = this._containerRef.createComponent(componentFactory);
+
+		component.instance.model = this._context.notebook as NotebookModel;
+		component.instance.cellModel = cell;
+
+		component.instance.handleContentChanged();
+
+		const element: HTMLElement = component.instance.outputRef.nativeElement;
+
+		const scale = element.clientWidth / screenshotWidth;
+		const canvasWidth = element.clientWidth / scale;
+		const canvasHeight = element.clientHeight / scale;
+
+		return toJpeg(component.instance.outputRef.nativeElement, { quality: .6, canvasWidth, canvasHeight, backgroundColor });
+	}
+}
+
+registerThemingParticipant((theme: IColorTheme, collector: ICssStyleCollector) => {
+	collector.addRule(`
+		#insert-dialog-cell-grid .loading-spinner-container {
```

Comment:
Non-dynamic CSS values should go in a separate .css file that's imported by this module